training/src/app/modules/chuck-norris/components/search/search.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, Output, EventEmitter, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { debounceTime, Subject, takeUntil, tap } from "rxjs";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss'],
+
+})
+export class SearchComponent implements OnInit {
+
+	public searchForm: FormGroup;
+	private searchKeyIsValid: boolean;
+	private componentDestroyed$: Subject<void>                        = new Subject();
+	@Output() private generateJokesOnClickMode: EventEmitter<string>  = new EventEmitter<string>();
```

Comment:
why do we need this event? the search component doesnt know that there is a click button somewhere.
maybe the name is not good? what is the purpose